org-gcal.el

```diff
@@ -147,15 +152,18 @@ to non-nil to inhibit notifications."
           (find-file-noselect (cdr i))
         (org-gcal--archive-old-event))))
   (dolist (cal org-gcal-file-alist)
-    (let ((b (oauth2-url-retrieve
-              (org-gcal-auth)
-              (format "%s?%s"
-                      (format org-gcal-events-url (first cal))
-                      (format "singleEvents=True&orderBy=startTime&timeMin=%s&timeMax=%s"
-                              (org-gcal--subtract-time)
-                              (org-gcal--add-time)))
-              'rgc-cb
-              (list cal skip-export)))))))
+    (make-thread (lambda ()
```

Comment:
Emacs 26.1 only ?

Just an idea, but how about carrying extra variables as
symbol-plist of calendar-id or something ?

Reply:
nah, the make-thread was just a crazy test. I guess this branch is turning into an 'experimental' branch. sorry for the confusion.

I don't fully get you on the plist for the calendar-id. What would it solve?

The remaining problem here is that:
- oauth2.el doesn't refresh the token automatically if used asynchronously.
- using synchronous requests works, but for multiple calendars it can become a little sluggish.
- as an "extra" annoyance, oauth2 uses plstore, and it looks like you have to enter the passphrase to decrypt the data of the tokens way to often and way too many times. This gives a shitty experience also.

I've been looking at google-contacts.el, calfw.el, and other similar packages and all of them have hacks around those.  I'll revert the thread thing, but not sure what's the next thing to try :(.  Feel free to throw ideas.